layer.md

```diff
@@ -166,31 +166,29 @@ The resulting tar archive for `rootfs-c9d-v1.s1` has the following entries:
 ./etc/.wh.my-app-config
 ```
 
-Where the basename name of `./etc/my-app-config` is now prefixed with `.wh.`, and will therefore be removed when the changeset is applied.
+where the basename name of `./etc/my-app-config` is now prefixed with `.wh.`, and will therefore be removed when the changeset is applied.
 
-## Applying
+## Applying Changesets
 
-Layer Changesets of [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` are applied rather than strictly extracted in normal fashion for tar archives.
+Layer Changesets of [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` are _applied_, rather than simply extracted as tar archives.
 
-Applying a layer changeset requires consideration for the [whiteout](#whiteouts) files.
-In the absence of any [whiteout](#whiteouts) files in a layer changeset, the archive is extracted like a regular tar archive.
+Applying a layer changeset requires special consideration for the [whiteout](#whiteouts) files.
 
+In the absence of any [whiteout](#whiteouts) files in a layer changeset, the archive is extracted like a regular tar archive.
 
 ### Changeset over existing files
 
-This section covers applying an entry in a layer changeset, if the file path already exists.
+This section describes applying an entry from a layer changeset if the file path already exists.
 
-If the file path is a directory, then the existing path just has it's attribute set from the layer changeset for that filepath.
-If the file path is any other file type (regular file, FIFO, etc), then the:
-* file path is unlinked (See [`unlink(2)`](http://linux.die.net/man/2/unlink))
-* create the file
-    * If a regular file then content written.
-* set attributes on the filepath
+If the entry and the existing path are both directories, then the existing path's attributes MUST be replaced by those of the entry in the changeset.
+In all other cases, the implementation MUST do the semantic equivalent of the following:
+- removing the file path (e.g. [`unlink(2)`](http://linux.die.net/man/2/unlink) on Linux systems)
```

Comment:
“removing the file” → “(recursively) removing the file”?  Most of the discussion in [the GNU docs you linked][1] is for “folks would be mad if we blew away a full tree and replaced it with a broken symlink”.  That makes sense for working filesystems, but we're building the rootfs from scratch here, so losing information is not a concern.  However, you don't want folks blindly using `unlink`, having it fail because the target is a directory and either dying or leaving it in place.  We *do* want them to get that old thing off the filesystem (whatever it takes ;), and the “(recursively)” hint may be enough to get that across.

Or maybe it's sufficient as it stands :p.

[1]: https://www.gnu.org/software/tar/manual/html_node/Dealing-with-Old-Files.html